evennia/contrib/worldloader/command.py

```diff
@@ -0,0 +1,281 @@
+           self.cmdset - the cmdset from which this command was picked. Not
+                         often used (useful for commands like 'help' or to
+                         list all available commands etc)
+           self.obj - the object on which this command was defined. It is often
+                         the same as self.caller.
+        """
+        pass
+
+    def func(self):
+        """
+        This is the hook function that actually does all the work. It is called
+         by the cmdhandler right after self.parser() finishes, and so has access
+         to all the variables defined therein.
+        """
+        self.caller.msg("Command called!")
+
+    def at_post_cmd(self):
+        """
+        This hook is called after self.func().
+        """
+        pass
+
+
+class MuxCommand(default_cmds.MuxCommand):
+    """
+    This sets up the basis for a Evennia's 'MUX-like' command
+    style. The idea is that most other Mux-related commands should
+    just inherit from this and don't have to implement parsing of
+    their own unless they do something particularly advanced.
+
+    A MUXCommand command understands the following possible syntax:
+
+      name[ with several words][/switch[/switch..]] arg1[,arg2,...] [[=|,] arg[,..]]
+
+    The 'name[ with several words]' part is already dealt with by the
+    cmdhandler at this point, and stored in self.cmdname. The rest is stored
+    in self.args.
+
+    The MuxCommand parser breaks self.args into its constituents and stores them
+    in the following variables:
+      self.switches = optional list of /switches (without the /)
+      self.raw = This is the raw argument input, including switches
+      self.args = This is re-defined to be everything *except* the switches
+      self.lhs = Everything to the left of = (lhs:'left-hand side'). If
+                 no = is found, this is identical to self.args.
+      self.rhs: Everything to the right of = (rhs:'right-hand side').
+                If no '=' is found, this is None.
+      self.lhslist - self.lhs split into a list by comma
+      self.rhslist - list of self.rhs split into a list by comma
+      self.arglist = list of space-separated args (including '=' if it exists)
+
+      All args and list members are stripped of excess whitespace around the
+      strings, but case is preserved.
+      """
+
+    def func(self):
+        """
+        This is the hook function that actually does all the work. It is called
+        by the cmdhandler right after self.parser() finishes, and so has access
+        to all the variables defined therein.
+        """
+        # this can be removed in your child class, it's just
+        # printing the ingoing variables as a demo.
+        super(MuxCommand, self).func()
+
+
+#------------------------------------------------------------
+# import csv tables
+#------------------------------------------------------------
+class CmdImportCsv(MuxCommand):
```

Comment:
For a contrib, the commands should probably be broken out into their own module without the default commands being in there. Makes the module easier to read - the commands are to be imported into the actual default anyway.